registry_module.go

```diff
@@ -265,6 +286,41 @@ func (r *registryModules) Create(ctx context.Context, organization string, optio
 	return rm, nil
 }
 
+func (r *registryModules) Update(ctx context.Context, moduleID RegistryModuleID, options RegistryModuleUpdateOptions) (*RegistryModule, error) {
+	if err := moduleID.valid(); err != nil {
+		return nil, err
+	}
+
+	if moduleID.RegistryName == "" {
+		log.Println("[WARN] Support for using the RegistryModuleID without RegistryName is deprecated as of release 1.5.0 and may be removed in a future version. The preferred method is to include the RegistryName in RegistryModuleID.")
+		moduleID.RegistryName = PrivateRegistry
+	}
+
+	if moduleID.RegistryName == PrivateRegistry && strings.TrimSpace(moduleID.Namespace) == "" {
+		log.Println("[WARN] Support for using the RegistryModuleID without Namespace is deprecated as of release 1.5.0 and may be removed in a future version. The preferred method is to include the Namespace in RegistryModuleID.")
+		moduleID.Namespace = moduleID.Organization
+	}
+
+	org := url.QueryEscape(moduleID.Organization)
+	registryName := url.QueryEscape(string(moduleID.RegistryName))
+	namespace := url.QueryEscape(moduleID.Namespace)
+	name := url.QueryEscape(moduleID.Name)
+	provider := url.QueryEscape(moduleID.Provider)
+	url := fmt.Sprintf("organizations/%s/registry-modules/%s/%s/%s/%s", org, registryName, namespace, name, provider)
```

Comment:
💅 A small style suggestion - in this file (& the rest of the code), we generally avoid creating variables for each URL element like on this [line](https://github.com/hashicorp/go-tfe/blob/3b3a69e7db8ecd4f3e36531b51d3cfd926420e5b/registry_module.go#L278).